backend/src/api/api.py

```diff
@@ -511,25 +512,25 @@ def add_package(
 
 
 @dataclass
-class Iterator(Generic[I]):
-    iter_supplier: Callable[[], Iterable[I | Exception]]
+class Generator(Generic[I]):
+    gen_supplier: Callable[[], typing.Generator[I | Exception, None, None]]
     expected_length: int
     fail_fast: bool = True
 
     @staticmethod
     def from_iter(
-        iter_supplier: Callable[[], Iterable[I | Exception]],
+        gen_supplier: Callable[[], typing.Generator[I | Exception, None, None]],
```

Comment:
Maybe we should call this just `supplier`. Same for the field.

Reply:
Sure